frontend/windows/handle_container.go

```diff
@@ -0,0 +1,124 @@
+package windows
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend"
+	"github.com/moby/buildkit/client/llb"
+	"github.com/moby/buildkit/frontend/dockerui"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const (
+	defaultBaseImage = "mcr.microsoft.com/windows/nanoserver:1809"
+	windowsSystemDir = "/Windows/System32/"
+)
+
+var (
+	defaultPlatform = ocispecs.Platform{
+		OS:           outputKey,
+		Architecture: runtime.GOARCH,
+	}
+)
+
+func handleContainer(ctx context.Context, client gwclient.Client, spec *dalec.Spec) (gwclient.Reference, *dalec.DockerImageSpec, error) {
+	sOpt, err := frontend.SourceOptFromClient(ctx, client)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if err := validateRuntimeDeps(spec); err != nil {
+		return nil, nil, fmt.Errorf("error validating windows spec: %w", err)
+	}
+
+	bc, err := dockerui.NewClient(client)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	targetPlatform, err := getTargetPlatform(bc)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pg := dalec.ProgressGroup("Build windows container: " + spec.Name)
+	worker := workerImg(sOpt, pg)
+
+	bin, err := buildBinaries(spec, worker, sOpt)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to build binary %w", err)
+	}
+
+	baseImgName := getBaseOutputImage(spec, targetKey, defaultBaseImage)
+	baseImage := llb.Image(baseImgName, llb.Platform(targetPlatform))
+
+	out := baseImage.
+		File(llb.Copy(bin, "/", windowsSystemDir)).
+		With(copySymlinks(spec.GetSymlinks(targetKey)))
+
+	def, err := out.Marshal(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	res, err := client.Solve(ctx, gwclient.SolveRequest{
+		Definition: def.ToPB(),
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	base := frontend.GetBaseOutputImage(spec, targetKey, defaultBaseImage)
+
+	img, err := frontend.BuildImageConfig(ctx, client, spec, targetKey, base, frontend.WithPlatform(targetPlatform))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ref, err := res.SingleRef()
+
+	return ref, img, err
+}
+
+func copySymlinks(lm map[string]dalec.SymlinkTarget) llb.StateOption {
+	return func(s llb.State) llb.State {
+		if len(lm) == 0 {
+			return s
+		}
+		keys := dalec.SortMapKeys(lm)
+		for _, srcPath := range keys {
+			l := lm[srcPath]
+			dstPath := l.Path
+			s = s.File(llb.Copy(s, srcPath, dstPath))
+		}
+
+		return s
+	}
+
+}
+
+func getTargetPlatform(bc *dockerui.Client) (ocispecs.Platform, error) {
+	platform := defaultPlatform
+
+	switch len(bc.TargetPlatforms) {
+	case 0:
+	case 1:
+		platform = bc.TargetPlatforms[0]
+	default:
+		return ocispecs.Platform{}, fmt.Errorf("multiple targets for a windows build, only amd64 is supported")
```

Comment:
This error mentions amd64 but doesn't validate it.
Also I'm not sure why it would matter since we aren't trying to run anything.

Reply:
We are making sure there are not multiple platforms specified. Because only one platform is supported (`windows/amd64`), if there is more than one platform requested by the user, the request cannot be satisfied.

The `amd64` mention was an attempt at providing additional information. The idea was to tell the user, "try again, but only with amd64".

I improved the error message. LMK if you want something different.

Reply:
done unless you want me to change it again.